### DISCUSSION TITLE
React Server Components Roadmap for Next.js 

### DISCUSSION THREAD

**Discussion by leerob** [vercel/next.js#31263] [App Router]
**Edit:** As of Next.js 13, Server Components are available to try out in the `app` directory (beta): https://nextjs.org/13

**Comment by mi3lix9** [comment]
Will RSC support SSG 👀

**Reply by shuding** [reply]
It will certainly support SSG! We’re still exploring the best possible way to do that. We will share when it’s more ready.

**Comment by jimbo** [comment]
Will middleware work with server components? (Or perhaps it already does?)

**Reply by leerob** [reply]
You will be able to use Middleware in conjunction with Server Components, yes. Middleware would run before a Server Component would execute.

**Comment by dawidk92** [comment]
Will hook useContext work with Server Components in the future?

**Comment by dawidk92** [comment]
Also, will Server Components work with Preact or will it work exclusively with React?

**Reply by faradaytrs** [reply]
It depends on preact

**Comment by jaredcbaum** [comment]
I'd love to be able to use a css-in-js library like styled components. I didn't open a feature request because it seems like there's plenty to do before that becomes a priority, like basic CSS Module support for example.

I cant exactly picture how this would work, I'm having trouble wrapping my head around how context would be handled between server and client components.

**Comment by ahmed-adly-khalil** [comment]
When beta is expected to arrive?

**Reply by bastienrobert** [reply]
https://github.com/vercel/next.js/discussions/35843#discussioncomment-2499879:
> You can follow the public roadmap on Github here: https://github.com/orgs/vercel/projects/39
